Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -156,6 +177,41 @@ fileprivate class ParameterReplacementVisitor: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  override func visit(_ node: GenericArgumentClauseSyntax) -> SyntaxVisitorContinueKind {
+    return .visitChildren
+  }
+
+  override func visit(_ node: GenericArgumentListSyntax) -> SyntaxVisitorContinueKind {
+    return .visitChildren
+  }
+
+  override func visit(_ node: GenericArgumentSyntax) -> SyntaxVisitorContinueKind {
+    let baseName = node.argument
+    guard let genericParameterClause = macro.genericParameterClause else {
+      return .skipChildren
+    }
+
+    let matchedParameter = genericParameterClause.parameters.enumerated().first { (index, parameter) in
+      return parameter.name.text == "\(baseName)"
```

Comment:
I think this will fail if you add spaces around `node.argument` because `"\(baseName)"` contains trivia while `parameter.name.text` does not.

You should probably do something like

```swift
guard let baseName node.argument.as(IdentifierTypeSyntax.self)?.name else {
  // Handle error
}
```

and then down here `baseName.text`.

Ie. I would expect a failing test case to be the following (both the space after `A` and the comment after `B` should trigger the failure.

```swift
macro gen<A , B /* some comment */>(a: A, b: B) = #otherMacro<A, B>(first: a, second: b)
```

Reply:
Weirdly enough this seems to just work already, but I'll adjust to use the `as ` as you suggested

Reply:
Done, thanks  -- added tests too

Reply:
wasn't sure if to error or just return, I think we can skip children here and call it a day